test/unit/app/browser/reducers/windowsReducerTest.js

```diff
@@ -20,36 +21,192 @@ describe('windowsReducer unit test', function () {
     maybeCreateWindow: (state, action) => state
   }
 
+  const fakeWindowApi = {
+    createWindow: () => {}
+  }
+
+  const fakePlatformUtil = {
+    isDarwin: () => true,
+    isWindows: () => false
+  }
+
   const state = Immutable.fromJS({
     windows: [],
     defaultWindowParams: {}
   })
-
+  let fakeTimers
   before(function () {
     mockery.enable({
       warnOnReplace: false,
       warnOnUnregistered: false,
       useCleanCache: true
     })
+    fakeTimers = sinon.useFakeTimers()
     mockery.registerMock('electron', fakeElectron)
     mockery.registerMock('ad-block', fakeAdBlock)
     mockery.registerMock('../../common/state/windowState', fakeWindowState)
+    mockery.registerMock('../windows', fakeWindowApi)
+    mockery.registerMock('../../common/lib/platformUtil', fakePlatformUtil)
     windowsReducer = require('../../../../../app/browser/reducers/windowsReducer')
   })
 
   after(function () {
     mockery.disable()
+    fakeTimers.restore()
   })
 
-  describe('APP_WINDOW_UPDATED', function () {
+  describe('APP_NEW_WINDOW', function () {
     let spy
+    before(function () {
+      spy = sinon.spy(fakeWindowApi, 'createWindow')
+    })
+    afterEach(function () {
+      spy.reset()
+    })
+    after(function () {
+      spy.restore()
+    })
 
+    const sampleFrame1 = {
+      location: 'http://mysite.com'
+    }
+    const sampleFrame2 = {
+      location: 'http://mysite2.com'
+    }
+
+    it('creates a window, with a single specified frame', function () {
+      const action = {
+        actionType: appConstants.APP_NEW_WINDOW,
+        frameOpts: sampleFrame1
+      }
+      windowsReducer(state, action)
+      fakeTimers.tick(0)
+      // ensure the window api was asked to create the frame
+      const actualCreateWindowFrameArg = spy.args[0][3]
```

Comment:
this is an interesting way to check the params- any reason you'd do this over something like `assert(spy.withArgs(args_here).calledOnce)`?

Reply:
@bsclifton In this case, sure that more expressive function could be used. I was just being consistent with the other tests here, which need to inspect the windowOptions object argument for property values, and not necessarily enforce every property on the object argument. If there's a nicer way to do that vs being tied to the argument order I'd love to learn!

Reply:
@bsclifton I updated the couple of tests I could using that syntax. It looks nicer, thanks. Only downside is the error messages aren't as expressive as using the chai.assert functions.